junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertionUtils.java

```diff
@@ -147,9 +145,7 @@ static boolean objectsAreEqual(Object obj1, Object obj2) {
 		if (obj1 == null) {
 			return (obj2 == null);
 		}
-		else {
-			return obj1.equals(obj2);
-		}
+		return obj1.equals(obj2);
```

Comment:
Oh, perhaps!

I do seem to vaguely remember though that, some time ago, somewhere in the JUnit 5 codebase, that `Objects.equals()` was used somewhere in the JUnit 5 code base, but it was realised that it was the cause of a bug, and so it was swapped out for correctly-functioning equivalence code.

I cannot remember if it was this method where the bug was found. So I'll do a quick search of the GitHub history, and if I can't find anything to do with this bug, then I'll try swapping the code out for `Objects.equals()` and see what happens.